src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -1144,4 +1045,213 @@ trait BronzeTransforms extends SparkSessionWrapper {
+                                                                    organizationId,
+                                                                    auditLogConfig,
+                                                                    apiEnv
+                                                                    )
+
+      if (rawSystemTableFiltered.isEmpty) {
+        throw new Exception(s"No data found in system.access.audit for organizationId: $organizationId " +
+          s"and fromTime: $fromTimeSysTableCompatible and untilTime: $untilTimeSysTableCompatible")
+      }
+
+      val isSqlEndpointEmpty = auditLogConfig.sqlEndpoint.getOrElse("").isEmpty
+      // if sql endpoint is not empty, the audit log data will be fetched from
+      // internal system tables then we need to convert the requestParams to json
+      val deriveRequestParams = if(isSqlEndpointEmpty) to_json(col("requestParams")) else col("requestParams")
+
+
+      val auditLogFromSysTable = SchemaTools.snakeToCamel(rawSystemTableFiltered)
+          .withColumn("organization_id", col("workspaceID"))
+          .withColumnRenamed("eventDate", "date")
+          .withColumn("timestamp", (col("eventTime").cast("double") * 1000).cast("long"))
+          .withColumn("requestParamsString", deriveRequestParams)
+          .drop("requestParams", "eventTime")
+
+      // if sql endpoint is not empty, the audit log data will be derived from extenal system tables
+      // then userIdentity and response cols needs to be converted to struct from string
+      val deriveUserIdentity = if(isSqlEndpointEmpty) col("userIdentity")
+        else structFromJson(spark, auditLogFromSysTable, "userIdentity")
+      val deriveResponse = if(isSqlEndpointEmpty) col("response")
+        else structFromJson(spark, auditLogFromSysTable, "response")
+
+      val auditLogFromSysTableToStruct = auditLogFromSysTable
+          .withColumn("requestParams", structFromJson(spark, auditLogFromSysTable, "requestParamsString"))
+          .withColumn("userIdentity", deriveUserIdentity)
+          .withColumn("response", deriveResponse)
+          .withColumn("hashKey", xxhash64('organization_id, 'timestamp, 'serviceName, 'actionName, 'requestId, 'requestParamsString))
+          .verifyMinimumSchema(Schema.auditMasterSchema)
+          .drop("requestParamsString")
+          .withColumn("response", $"response".withField("statusCode",
+            coalesce($"response.statusCode", $"response.status_code".cast(LongType))))
+          .withColumn("response", $"response".withField("errorMessage",
+            coalesce($"response.errorMessage", $"response.error_message")))
+         .withColumn("response", struct($"response.statusCode", $"response.errorMessage", $"response.result"))
+
+      auditLogFromSysTableToStruct
+  } catch {
+      case e: org.apache.spark.sql.AnalysisException =>
+        throw new Exception(s"Access issue with table system.access.audit: ${e.getMessage}")
```

Comment:
At this point the error is while parsing the audit logs, not accessing them, let's make this error a bit more helpful

Reply:
done